sky/spot/recovery_strategy.py

```diff
@@ -295,9 +300,7 @@ def _launch(self,
                     exceptions.ResourcesMismatchError) as e:
                 logger.error('Failure happened before provisioning. '
                              f'{common_utils.format_exception(e)}')
-                if raise_on_failure:
```

Comment:
Left a comment on signature on whether this is ok.

Reply:
This should be raised directly, as retry will not help with these issues.